src/gui/progressbardelegate.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020  Prince Gupta <jagannatharjun11@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "progressbardelegate.h"
+
+#include <QApplication>
+#include <QModelIndex>
+#include <QPainter>
+#include <QStyleOptionViewItem>
+
+#if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
+#include <QProxyStyle>
+#endif
+
+ProgressBarDelegate::ProgressBarDelegate(const int progressColumn, const int dataRole, QObject *parent)
+    : QStyledItemDelegate {parent}
+    , m_progressColumn {progressColumn}
+    , m_dataRole {dataRole}
+{
+#if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
+    m_dummyProgressBar.setStyle(new QProxyStyle {"fusion"});
+#endif
+}
+
+void ProgressBarDelegate::initProgressStyleOption(QStyleOptionProgressBar &option, const QModelIndex &index) const
+{
+    option.text = index.data().toString();
+    option.progress = static_cast<int>(index.data(m_dataRole).toReal());
+    option.maximum = 100;
+    option.minimum = 0;
+    option.state |= (QStyle::State_Enabled | QStyle::State_Horizontal);
```

Comment:
this should fix https://github.com/qbittorrent/qBittorrent/issues/12271 https://github.com/qbittorrent/qBittorrent/issues/13550

can't test it though, already have two VM installed on my computer can't run another, low on disk space.
maybe someone from those issues can test it (@comicallycluttered @thalieht  ?)

Reply:
>this should fix #12271 #13550

Indeed, tested but i don't think it's this line (commented out and it made no difference).
Dunno if this was mentioned in the issues but the Done column is still excluded (visually) when selecting a torrent in the list (white background).

Reply:
@thalieht thanks for test, this should fix visual exclusion (at least it did on my stylesheet)

```diff
diff --git a/src/gui/progressbardelegate.cpp b/src/gui/progressbardelegate.cpp
index 706f80799..49cf82551 100644
--- a/src/gui/progressbardelegate.cpp
+++ b/src/gui/progressbardelegate.cpp
@@ -67,7 +67,25 @@ void ProgressBarDelegate::paint(QPainter *painter, const QStyleOptionViewItem &o
     newopt.rect = option.rect;
     initProgressStyleOption(newopt, index);

+    QStyle *widgetStyle = option.widget ? option.widget->style() : qApp->style();
+
     painter->save();
+    widgetStyle->drawPrimitive(QStyle::PE_PanelItemViewItem, &option, painter, option.widget);
+
+    // draw the focus rect
+    if (option.state & QStyle::State_HasFocus) {
+        QStyleOptionFocusRect o;
+        o.QStyleOption::operator=(option);
+        o.rect = widgetStyle->subElementRect(QStyle::SE_ItemViewItemFocusRect, &option, option.widget);
+        o.state |= QStyle::State_KeyboardFocusChange;
+        o.state |= QStyle::State_Item;
+        QPalette::ColorGroup cg = (option.state & QStyle::State_Enabled)
+                                      ? QPalette::Normal : QPalette::Disabled;
+        o.backgroundColor = option.palette.color(cg, (option.state & QStyle::State_Selected)
+                                                        ? QPalette::Highlight : QPalette::Window);
+        widgetStyle->drawPrimitive(QStyle::PE_FrameFocusRect, &o, painter, option.widget);
+    }
+
     m_dummyProgressBar.style()->drawControl(QStyle::CE_ProgressBar, &newopt, painter, &m_dummyProgressBar);
     painter->restore();
 }
```

Reply:
How about this?
```diff
diff --git a/src/gui/progressbardelegate.cpp b/src/gui/progressbardelegate.cpp
index 706f80799..9749e481b 100644
--- a/src/gui/progressbardelegate.cpp
+++ b/src/gui/progressbardelegate.cpp
@@ -68,6 +68,14 @@ void ProgressBarDelegate::paint(QPainter *painter, const QStyleOptionViewItem &o
     initProgressStyleOption(newopt, index);

     painter->save();
+
+    QStyleOptionViewItem o;
+    o.QStyleOption::operator=(option);
+    o.text = QString {};
+    o.icon = QIcon {};
+    QStyle *widgetStyle = option.widget ? option.widget->style() : qApp->style();
+    widgetStyle->drawControl(QStyle::CE_ItemViewItem, &o, painter, option.widget);
+
     m_dummyProgressBar.style()->drawControl(QStyle::CE_ProgressBar, &newopt, painter, &m_dummyProgressBar);
     painter->restore();
 }
```

Reply:
Both fix it 👍

Reply:
This might be a problem though

![Capture](https://user-images.githubusercontent.com/6451685/97355023-f8066480-189e-11eb-8a54-d3cce011b779.JPG)

Reply:
try this
```diff
diff --git a/src/gui/progressbardelegate.cpp b/src/gui/progressbardelegate.cpp
index 706f80799..b698c9df2 100644
--- a/src/gui/progressbardelegate.cpp
+++ b/src/gui/progressbardelegate.cpp
@@ -43,6 +43,7 @@ ProgressBarDelegate::ProgressBarDelegate(const int progressColumn, const int dat
     , m_dataRole {dataRole}
 {
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
+    // XXX: To avoid having the progress text on the right of the bar
     m_dummyProgressBar.setStyle(new QProxyStyle {"fusion"});
 #endif
 }
@@ -65,9 +66,20 @@ void ProgressBarDelegate::paint(QPainter *painter, const QStyleOptionViewItem &o
     QStyleOptionProgressBar newopt;
     newopt.initFrom(&m_dummyProgressBar);
     newopt.rect = option.rect;
+    newopt.state = option.state;
     initProgressStyleOption(newopt, index);

     painter->save();
+
+    // draw background and focus rect
+    QStyleOptionViewItem o;
+    o.QStyleOption::operator=(option);
+    o.text = QString {};
+    o.icon = QIcon {};
+    const QStyle *style = option.widget ? option.widget->style() : QApplication::style();
+    style->drawControl(QStyle::CE_ItemViewItem, &o, painter, option.widget);
+
+    // draw progressbar
     m_dummyProgressBar.style()->drawControl(QStyle::CE_ProgressBar, &newopt, painter, &m_dummyProgressBar);
     painter->restore();
```

Reply:
👍 

![Capture2](https://user-images.githubusercontent.com/6451685/97356787-7106bb80-18a1-11eb-878b-39cbff2e677a.JPG)

Reply:
ok last thing add this after 
   newopt.QStyleOption::operator=(option);

after 
QStyleOptionProgressBar newopt;

then it should fix the text color too

Reply:
>then it should fix the text color too

Uhh fix what? I don't see anything wrong... anyway this doesn't change anything.

Reply:
> nvm i just saw everything else is white on selected line...

are you willing to fix it? maybe you can open a separate PR with these changes?

Reply:
>are you willing to fix it? maybe you can open a separate PR with these changes?

No way, I don't know what all this stuff does.

Reply:
np, it's enough for now. @Chocobo1 please merge this.